### DISCUSSION TITLE
creating symlink under /usr/lib

### DISCUSSION THREAD

**Discussion by ChipHunter** [OE4T/tegra-demo-distro#123] [Q&A] [answered]
Hi All, 

First of all, Thanks a lot for the great work!
 
We would like to use the tegra-demo-destro in our product, so I have been working on migrating our application from using only meta-tegra to using this distro-demo. But since the rootfs is now read only, new directory structure is needed. I was trying to install our executable to the home directory along with the needed shared libraries (the deployment happens by copying all the needed files after the board boots). But since the home dir is not in the list of directories to search for shared libraries, I tried to create a symlink under /usr/lib to the location of the shared libs. I added the following to my image recipe: 

symlinkfunction() {
   
    ln -sf /home/root/my_app/lib/my_lib.so.21.4.10 ${IMAGE_ROOTFS}/usr/lib/my_lib.so.21.4
    
}

ROOTFS_POSTPROCESS_COMMAND += "symlinkfunction; "

This doesn't work for some reasons I don't understand. If I create the symlink anywhere else other than /lib or /usr/lib, it works without any issue. I am really struggling with this issue since few days now so your support would be very appreciated!

Thanks!

PS: The build system is not showing any errors. It seems either this function is discarded silently or is overwritten after it executes

**Comment by madisongh** [comment, marked as answer]
I think it's `ldconfig` (which is run after the rootfs is constructed) doing this to you.

Try making that symlink relative, rather than absolute, i.e.:
```
symlinkfunction() {

ln -sf ../../home/root/my_app/lib/my_lib.so.21.4.10 ${IMAGE_ROOTFS}/usr/lib/my_lib.so.21.4
}
```
That worked better in my tests.   Another alternative would be to have the recipe install a config file in `/etc/ld.so.conf.d` that contains `/home/root/my_app/lib`.  That will struct `ldconfig` to add that directory to the shared library search path.  Then you could do away with the symlink altogether.
